### DISCUSSION TITLE
Is there a way to disable on-typing assists?

### DISCUSSION THREAD

**Discussion by tomalexander** [joaotavora/eglot#1297] [Q&A] [answered]
Rust-analyzer has some [on-typing assists that will do things like automatically inserting closing parenthesis/braces when you type an opening parenthesis/brace](https://rust-analyzer.github.io/manual.html#on-typing-assists). Their documentation states that for vscode this can be controlled by editing `settings.json` and setting `editor.formatOnType`. As far as I can tell, that is a vscode-specific setting so I don't think I could toggle it with the initialization options since it does not appear in the [list of initialization options](https://rust-analyzer.github.io/manual.html#configuration). Does anyone know if eglot has an equivalent setting I can disable?

(The context is I'm writing a bunch of code using nom, which is a very parenthesis-heavy syntax, and this functionality is injecting parenthesis in incorrect locations which is brutal to debug when you're 10+ parenthesis levels deep.)

**Comment by joaotavora** [comment, marked as answer]
See variable eglot-ignored-server-capabilities

**Reply by tomalexander** [reply]
Thank you!

**Reply by joaotavora** [reply]
No problem. Sorry I was so terse, was on mobile.  The way is to add `:documentOnTypeFormattingProvider`to that variable's list.

**Reply by tomalexander** [reply]
No worries! I just needed to be pointed in the right direction, but thanks for the follow-up, that'll save me some time digging.
